library/core/src/alloc/global.rs

```diff
@@ -205,4 +205,44 @@ pub unsafe trait GlobalAlloc {
         }
         new_ptr
     }
+
+    /// Allocates a block of memory for `layout` which is guaranteed to contains
+    /// no GC pointers.
+    ///
+    /// An atomic allocation has the same semantics as regular `alloc` except
+    /// for one key difference which is that it tells the collector not to trace
+    /// this block during marking.
+    ///
+    /// # Safety
+    ///
+    /// The block must not contain any pointers which, either directly or
+    /// transitively, point to a GC'd value.
+    #[stable(feature = "global_alloc", since = "1.28.0")]
+    unsafe fn alloc_atomic(&self, layout: Layout) -> *mut u8;
+
+    /// Allocates a block of memory for `layout` which is traced conservatively
+    /// by the collector during marking.
+    #[stable(feature = "global_alloc", since = "1.28.0")]
+    fn alloc_conservative(&self, layout: Layout) -> *mut u8;
+
+    /// Allocates a block of memory for `layout` where word-aligned fields are
+    /// described to the collector by `bitmap`.
+    ///
+    /// An atomic allocation has the same semantics as regular `alloc` except
```

Comment:
There's some duplication of this comment above.

Reply:
I wonder if this should be s/atomic/precise/ ?

Reply:
copy / paste bug! Fixed here: bd6ddff72acc08d6f5baa0bef41a0c816cc9490c